packages/libs/eda/src/lib/core/components/visualizations/implementations/NetworkVisualization.tsx

```diff
@@ -307,6 +335,17 @@ function NetworkViz(props: VisualizationProps<Options>) {
     </div>
   );
 
+  // list of node labels
+  const labels = useMemo(() => {
```

Comment:
i dont follow - why is `labels` defined twice? Once here and once above?

Reply:
@asizemore Yes there are two `labels` variables, which are now changed to `nodeLabels` and they have different roles. The first one is defined inside `cleanedData` to set the default of the node label control to be "Select All" (all node labels are checked initially, which mimics "show all node labels in the network plot": since `cleanedData` is defined using useMemo hook with dependencies, it will be more efficient to set the default "Select All" in there. The second one is to pass the list of all node labels to node label control so that all node labels are shown in the drop down menu with checkboxes.

Reply:
thanks @moontrip ! That's very helpful. I made one further suggestion in the latest review about it.